rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/data/JobStateImpl.java

```diff
@@ -85,6 +88,19 @@ public String getOwner() {
     }
 
     @Override
+    public List<TaskState> getTaskByTag(String tag) {
+        ArrayList<TaskState> taskStateList = new ArrayList<>();
+        Map<String, TaskStateData> taskStateMap = d.getTasks();
+        for (TaskStateData ts : taskStateMap.values()) {
+            String taskTag = ts.getTag();
+            if(taskTag != null && taskTag.equals(tag)) {
```

Comment:
Instead of filtering tasks each time the method is called, is it not possible to introduce a new datastructure (e.g. a [Multimap](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Multimap.html) to keep TaskStateData indexed by tags for object that have a tag, thus avoiding to scan all entries?